docs/user-guide/diff-strategies.md

```diff
@@ -0,0 +1,90 @@
+# Diff Strategies
+
+Argo CD calculates the diff between the desired state and the live
+state in order to define if an Application is out-of-sync. This same
+logic is also used in Argo CD UI to display the differences between
+live and desired states for all resources belonging to an application.
+
+Argo CD currently has 3 different strategies to calculate diffs:
+
+- **Legacy**: This is the main diff strategy used by default. It
+  applies a 3-way diff based on live state, desired state and
+  last-applied-configuration (annotation).
+- **Structured-Merge Diff**: Strategy automatically applied when
+  enabling Server-Side Apply sync option. 
+- **Server-Side Diff**: New strategy that invokes a Server-Side Apply
+  in dryrun mode delegating to Kube-API the action of calculating the
+  diff.
```

Comment:
> delegating to Kube-API the action of calculating the diff.

I'm not sure that's entirely true... we still calculate the diff, we just use server-side apply to get the predicted live state.

Reply:
fair. fixed